controllers/restaurantController.js

```diff
@@ -0,0 +1,165 @@
+import { Restaurant } from "../models/Restaurant.js";
+import mongoose from "mongoose";
+
+const getRestaurants = async function (req, res) {
+  try {
+    const queryObj = { ...req.query };
+    let query = Restaurant.find(queryObj);
+    const restaurants = await query;
+
+    if (restaurants.length === 0) {
```

Comment:
mhhh, is 404 correct here? like this request was proceed but because of query we got empty data, right? do we consider empty result as client error ???